source/markdown/data_exploratie_2023.Rmd

```diff
@@ -0,0 +1,1596 @@
+  group_by(landbouwstreek, jaar, periode_in_jaar, naam) %>%
+  summarise(gemiddelde_freq = weighted.mean(relfreq, w)) %>%
+  ungroup() %>%
+  # Calculate frequency per region as unweighted average
+  group_by(landbouwstreek, jaar, naam) %>%
+  summarise(gemiddelde_freq = mean(gemiddelde_freq)) %>%
+  ungroup() %>%
+  # Add weights of regions to calculate average over regions
+  full_join(gewicht_per_landbouwstreek, by = join_by(landbouwstreek)) %>%
+  group_by(jaar, naam) %>%
+  mutate(order_var = weighted.mean(gemiddelde_freq, w)) %>%
+  ungroup() %>%
+  mutate(naam = reorder(naam, order_var,
+                        decreasing = FALSE)) %>%
+  select(-c(n, w)) %>%
+  pivot_wider(names_from = "landbouwstreek",
+              values_from = "gemiddelde_freq") %>%
+  select(naam, Leemstreek, Zandleemstreek, gewogen_gemiddelde = order_var) %>%
+  arrange(desc(gewogen_gemiddelde)) %>%
+  rownames_to_column(var = "rang") %>%
+  mutate(rang = as.numeric(rang))
+
+write_csv(relfreq_vogels_regio, here(media_folder, "relfreq_vogels_regio.csv"))
+  
+# Get 20 most frequent species or goal species
+relfreq_vogels_regio %>%
+  filter(rang <= 20 | naam %in% doelsoorten) %>%
+  kable(digits = 3)
+```
+
+In de volgende figuur is elk punt een relatieve frequentie per telperiode.
+
+```{r}
+# Calculate relative frequency per region per period as weighted mean
+freq_to_plot <- relfreqs_vogels %>%
+  group_by(landbouwstreek, jaar, periode_in_jaar, naam) %>%
+  summarise(gemiddelde_freq = weighted.mean(relfreq, w)) %>%
+  ungroup() %>%
+  full_join(gewicht_per_landbouwstreek) %>%
+  group_by(jaar, naam) %>%
+  mutate(order_var = weighted.mean(gemiddelde_freq, w)) %>%
+  ungroup() %>%
+  # Retrieve 20 most common species + goal species
+  mutate(naam = reorder(naam, order_var,
+                        decreasing = FALSE)) %>%
+  filter(as.numeric(naam) > length(unique(naam)) - 20 |
+           naam %in% doelsoorten)
+
+p <- freq_to_plot %>%
+  ggplot() +
+    geom_vline(xintercept = 0, alpha = 0.2, linetype = "dashed") +
+    geom_hline(yintercept = length(unique(freq_to_plot$naam)) - 20 + 0.5,
+               alpha = 0.2, linetype = "dashed") +
+    geom_point(aes(x = gemiddelde_freq, y = naam, colour = landbouwstreek),
+               size = 1) +
+    labs(y = "", x = "Relatieve frequentie", colour = "Regio") +
+    scale_y_discrete(labels = function(x) highlight(x, doelsoorten, "red")) +
+    theme(legend.position = "top",
+          axis.text = element_text(size = 8),
+          axis.text.y = element_markdown(),
+          panel.spacing = unit(0.7, "lines")) +
+    guides(colour = guide_legend(override.aes = list(size = 2)))
+
+save_figure(p, here(media_folder, "vogelfreqs_2023"), devices,
+            dpi = 300, width = 8, height = 6)
+
+p
+```
+
+Top 20 van soorten die in termen van verschil in gemiddelde relatieve frequentie van voorkomen meer voorkomen in de Zandleemstreek dan in de Leemstreek:
+
```

Comment:
1. Misschien ook interessant om niet enkel naar absoluut verschil te kijken, maar ook naar de ratio dus relatieve frequentie streek1 / relatieve frequentie streek2. De ringmus is bv ongeveer x3 in zandleemstreek (maar blijft wel zeldzaam, waardoor ze in de absolute verschil vergelijking uit de top 20 valt).

Reply:
Goed plan! merci

Reply:
via #12